zipkin-storage/elasticsearch/src/main/java/zipkin/storage/elasticsearch/InternalElasticsearchClient.java

```diff
@@ -48,7 +48,7 @@
   protected static final String DEPENDENCY_LINK = "dependencylink";
 
   protected interface Factory {
-    InternalElasticsearchClient create();
+    InternalElasticsearchClient create(String allIndices);
```

Comment:
I had to bring this back to support the index-query-rewriting thing I'm doing in the RestClient (see the javadoc attached to `MAX_INDICES`). It's definitely ugly, but IMO less so than making the client check the # of indices and do the query rewriting themselves.